k8s-client/listpods.go

```diff
@@ -18,18 +18,36 @@ limitations under the License.
 package main
 
 import (
+	"flag"
 	"fmt"
 	"time"
 
+	//	"github.com/kubernetes/kubernetes/staging/src/k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd"
 	//metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	mach_apis_meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	//	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
 	"k8s.io/client-go/kubernetes"
-	metav1 "k8s.io/client-go/pkg/api/v1"
 	"k8s.io/client-go/rest"
 )
 
+var (
+	master     = flag.String("master", "", "Master's URL to communicate with kube-apiserver if running from outside the cluster or if running kube-apiserver with insecure flag.")
+	kubeconfig = flag.String("kubeconfig", "", "Absolute path to kubeconfig if running from outside the cluster or if running apiserver with insecure flag.")
+)
+
 func main() {
+	flag.Parse()
 	// creates the in-cluster config
-	config, err := rest.InClusterConfig()
+	var config *rest.Config
+	var err error
+	if *master != "" || *kubeconfig != "" {
+		fmt.Printf("Either master or kubeconfig specified. building kube config from that..\n")
```

Comment:
Move this statement after actual execution.
The message can be "Client config was built using flags: Address: '%s' Kubeconfig: '%s' \n"